src/test/unit/scheduler/test_compare.py

```diff
@@ -32,58 +27,40 @@ def get_vfp_of_included_text_files(self, root_uid, blacklist=None):
         return {}
 
 
-class TestSchedulerCompare(unittest.TestCase):
-    @mock.patch('plugins.base.ViewUpdater', lambda *_: None)
-    def setUp(self):
-        self.config = ConfigParser()
-        self.config.add_section('expert-settings')
-        self.config.set('expert-settings', 'block-delay', '2')
-        self.config.set('expert-settings', 'ssdeep-ignore', '80')
-
-        db_mock = MockDbInterface()
-        self.bs_patch_new = unittest.mock.patch(
-            target='storage.binary_service.BinaryService.__new__', new=lambda *_, **__: db_mock
-        )
-        self.bs_patch_init = unittest.mock.patch(
-            target='storage.binary_service.BinaryService.__init__', new=lambda _: None
-        )
-        self.bs_patch_new.start()
-        self.bs_patch_init.start()
-
-        self.compare_scheduler = ComparisonScheduler(db_interface=db_mock, admin_db_interface=db_mock, testing=True)
-
-    def tearDown(self):
-        self.compare_scheduler.shutdown()
-        self.bs_patch_new.stop()
-        self.bs_patch_init.stop()
-
-        gc.collect()
-
-    def test_start_compare(self):
-        self.compare_scheduler.add_task(('existing_id', True))
-        uid, redo = self.compare_scheduler.in_queue.get(timeout=2)
+@pytest.mark.cfg_defaults(
+    {
+        'expert-settings': {
+            'ssdeep-ignore': 80,
+        },
+    },
+)
+@pytest.mark.SchedulerTestConfig(start_processes=False, comparison_db_class=MockDbInterface)
+class TestSchedulerCompare:
+    def test_start_compare(self, comparison_scheduler):
+        comparison_scheduler.add_task(('existing_id', True))
+        uid, redo = comparison_scheduler.in_queue.get(timeout=2)
         assert uid == 'existing_id', 'retrieved id not correct'
         assert redo, 'redo argument not correct'
 
-    def test_start(self):
-        self.compare_scheduler.start()
+    def test_start(self, comparison_scheduler):
+        comparison_scheduler.start()
```

Comment:
if `start_processes` is `False`, will `shutdown()` ever get called?

Reply:
Nope, good catch.